crates/router/src/core/payment_methods/cards.rs

```diff
@@ -386,7 +391,12 @@ pub async fn get_card_from_hs_locker<'a>(
     #[cfg(not(feature = "kms"))]
     let jwekey = &state.conf.jwekey;
     #[cfg(feature = "kms")]
-    let jwekey = &state.kms_secrets;
+    let jwekey = &state
```

Comment:
why are we making this change is it needed?

Reply:
This is the total change:
```
let jwekey = &state
        .kms_secrets
        .as_ref()
        .ok_or(errors::VaultError::MissingRequiredField {
            field_name: "kms_secrets",
        })?;
```
as we have made KMS optional, we need this change to pull out the option